File: moderate/split_the_number.py
#!/usr/bin/env python2
"""
- CodeEval - Split the Number
- https://www.codeeval.com/open_challenges/131/
"""

from sys import argv

def do_calc(n, pattern):
    if '+' in pattern:
        sign = '+'
        index_sign = pattern.index('+')
    else:
        sign = '-'
        index_sign = pattern.index('-')

    x, y = int( n[:index_sign] ), int( n[index_sign:] )
    if sign == '+':
        return x+y
    else:
        return x-y
    

def main():
    with open(argv[1]) as f:
        for line in f:
            n, pattern = line.strip().split(' ')
            result = do_calc(n, pattern)
            print(result)

if __name__ == "__main__":
    main()

File: easy/SplitTheNumber.java
/* CodeEval - Split the Number
 * https://www.codeeval.com/open_challenges/131/
 */

import java.io.*;

public class SplitTheNumber {
    
    static int doCalc(String n, String pattern){
        char sign = '+';        
        int indexOfSign  = pattern.indexOf(sign);
        if (indexOfSign < 0){
            sign = '-';
            indexOfSign  = pattern.indexOf(sign);
        }

        int x = Integer.parseInt(n.substring(0, indexOfSign));
        int y = Integer.parseInt(n.substring(indexOfSign, n.length()));

        if (sign == '+')
            return x+y;
        else
            return x-y;
    }

    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            // Process line of input Here
            String[] tokens = line.split(" ");
            int result = doCalc(tokens[0], tokens[1]);
            System.out.println(result);
        }
    }
}
